src/vellum/workflows/state/context.py

```diff
@@ -22,23 +25,32 @@ def __init__(
         vellum_client: Optional[Vellum] = None,
         execution_context: Optional[ExecutionContext] = None,
         generated_files: Optional[dict[str, str]] = None,
+        enable_monitoring: Optional[bool] = None,
     ):
         self._vellum_client = vellum_client
         self._event_queue: Optional[Queue["WorkflowEvent"]] = None
         self._node_output_mocks_map: Dict[Type[BaseOutputs], List[MockNodeExecution]] = {}
         self._execution_context = get_execution_context()
 
-        if execution_context is not None:
+        # Auto-enable monitoring for external workflows if VELLUM_API_KEY is set
+        if enable_monitoring is None:
+            enable_monitoring = bool(os.getenv("VELLUM_API_KEY"))
 
+        if execution_context is not None:
             self._execution_context.trace_id = execution_context.trace_id
-
             if execution_context.parent_context is not None:
                 self._execution_context.parent_context = execution_context.parent_context
 
+        # For external workflows, ensure proper trace_id and parent context
         if self._execution_context.parent_context is None:
             self._execution_context.parent_context = ExternalParentContext(span_id=uuid4())
 
+        # Auto-generate trace_id for external workflows if not set
+        if enable_monitoring and str(self._execution_context.trace_id) == "00000000-0000-0000-0000-000000000000":
```

Comment:
I ting we could remove the `enable_monitoring` - we'll always want to set the trace id if it's all 0